pkg/sca/sca.go

```diff
@@ -631,6 +631,11 @@ func generateSharedObjectNameDeps(ctx context.Context, hdl SCAHandle, generated
 			}
 
 			for _, soname := range sonames {
+				// Cuda is a dangling library, which must come from the host
```

Comment:
While the comment is accurate, I don't feel like it sufficiently explains why this code is here. I think the case is that we don't want to ship this library in the system path because it will conflict with the driver injected by the host, and having packages that provide the so: are footguns that make it easy to unintentionally do so. That may be more text than we want to add here, but maybe it could be in the commit message?